### PR TITLE
Fixes #31382 - Hostgroup content facet migration fix

### PR DESCRIPTION
**Pull request by parthaa**
Added unscoped when checking for anonymous_admin.

### CODE REVIEW

File: db/migrate/20150930183738_migrate_content_hosts.rb [hunk 1]
```diff
@@ -329,7 +329,7 @@ def update_hypervisor_consumer(hypervisor)
   # rubocop:disable Metrics/MethodLength
   # rubocop:disable Metrics/AbcSize
   def up
-    if User.where(:login => User::ANONYMOUS_API_ADMIN).first.nil?
+    if User.unscoped.where(:login => User::ANONYMOUS_API_ADMIN).first.nil?
```
Comment (ekohl) [inline db/migrate/20150930183738_migrate_content_hosts.rb:332]: Not needed, but shouldn't this also be `exists?` like the other?
```suggestion
    if User.unscoped.where(:login => User::ANONYMOUS_API_ADMIN).exists?
```

File: db/migrate/20150930183738_migrate_content_hosts.rb [hunk 2]
```diff
@@ -329,7 +329,7 @@ def update_hypervisor_consumer(hypervisor)
   # rubocop:disable Metrics/MethodLength
   # rubocop:disable Metrics/AbcSize
   def up
-    if User.where(:login => User::ANONYMOUS_API_ADMIN).first.nil?
+    if User.unscoped.where(:login => User::ANONYMOUS_API_ADMIN).exists?
```
Comment (parthaa) [inline db/migrate/20150930183738_migrate_content_hosts.rb:332]:
```suggestion
    unless User.unscoped.where(:login => User::ANONYMOUS_API_ADMIN).exists?
```

**Comment by chris1984** [comment]
@parthaa is there a way to reproduce this?

**Comment by parthaa** [comment]
> @parthaa is there a way to reproduce this?

- I installed 3.16
- created a hostgroup that has a cv, content source,  a lifecycle env, an os and arch (no repo id, but dont think it matters)
- upgraded to 3.17 using the instructions here -> https://theforeman.org/plugins/katello/nightly/upgrade/index.html (I used 3.17 instead of nightly.)

**Review by chris1984 — APPROVED** [review]
ACK works good:

Without the patch after upgrade:

https://nimb.ws/VkyTLu

With patch after upgrade:

https://nimb.ws/IiQWmY

**Comment by parthaa** [comment]
[test katello]

**Comment by parthaa** [comment]
[test katello]

**Comment by chris1984** [comment]
[test katello]

**Comment by parthaa** [comment]
[test katello]

**Comment by chris1984** [comment]
[test katello]

**Comment by parthaa** [comment]
[test katello]

**Pull request merged**